app/models/item_change_set.rb

```diff
@@ -47,6 +48,12 @@ def setup_properties!(_options)
     setup_view_access_with_cdl_properties(model.access)
   end
 
+  def format_of_catalog_record_ids
+    return if catalog_record_ids.blank? || CatalogRecordId.valid?(catalog_record_ids)
+
+    errors.add(:catalog_record_ids, "are not a valid format")
+  end
+
   def sync
```

Comment:
this adds validation of catalog record IDs to the item change set (it was missing before) ...

the logic difference between folio vs symphony already exists and is abstracted away inside the `CatalogRecordId.valid?` method

Note: This adds some extra validation to the `ItemChangeSet` which is used by other parts of the code.  I believe this is OK because as soon as you save the change set, the catalog record ID would be validated anyway, and would throw an error if invalid on save.  So I don't think there are places in the code that would have worked ok that will now fail this addition (i.e. because a change set validation should be followed by a save anyway).